src/GraphQL.Serialization.Newtonsoft/Http/DocumentWriter.cs

```diff
@@ -0,0 +1,49 @@
+using System.Buffers;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace GraphQL.Http
+{
+    public class DocumentWriter : IDocumentWriter
+    {
+        private readonly JsonArrayPool _jsonArrayPool = new JsonArrayPool(ArrayPool<char>.Shared);
+        private readonly JsonSerializer _serializer;
+        internal static readonly Encoding Utf8Encoding = new UTF8Encoding(false);
+
+        public DocumentWriter()
+            : this(indent: false)
+        {
+        }
+
+        public DocumentWriter(bool indent)
+            : this(
+                indent ? Formatting.Indented : Formatting.None,
+                new JsonSerializerSettings())
+        {
+        }
+
+        public DocumentWriter(Formatting formatting, JsonSerializerSettings settings)
+        {
+            _serializer = JsonSerializer.CreateDefault(settings);
+            _serializer.Formatting = formatting;
+            _serializer.ContractResolver = new NewtonsoftContractResolver(_serializer.ContractResolver);
+        }
+
+        public async Task WriteAsync<T>(Stream stream, T value)
+        {
+            using (var writer = new HttpResponseStreamWriter(stream, Utf8Encoding))
```

Comment:
I think both the `JsonArrayPool` and `HttpResponseStreamWriter` classes need to also be moved to this library.

Reply:
Agree

Reply:
Moved.